agent/lm_agent/utils.py

```diff
@@ -17,6 +18,10 @@ async def run_command(command_line: str):
     output = str(stdout, encoding=ENCODING)
 
     if proc.returncode != 0:
-        logger.error(f"Error: {output} | Return Code: {proc.returncode}")
-        return None
+        logger.error(
+            f"Command {command_line} failed!",
+            f"Error: {output}",
+            f"Return code: {proc.returncode}",
+        )
+        return False
```

Comment:
I'm wondering if it would be better to raise an exception here instead. It would make it clearer when there is an error condition. This is especially true since the docstring doesn't explain that False indicates failure. It would be nicer if you could trust that any output from a function is valid and an error condition always results in an exception.

There could be some confusion if the output is empty, for example, but the command does not fail. In this case, the return value would still be falsey and fail the tests you had earlier, but there would be no log record indicating what the error was.

Reply:
Yes, I refactored the function as you suggested:

- Returns the output as string if the command succeeds.
- Raises CommandFailedToExecute exception if return code is not zero.